src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenAsyncEHTests.cs

```diff
@@ -287,10 +287,10 @@ .maxstack  3
           IL_0093:  call       ""void System.Runtime.CompilerServices.YieldAwaitable.YieldAwaiter.GetResult()""
           IL_0098:  ldarg.0
           IL_0099:  ldarg.0
-          IL_009a:  ldfld      ""int Test.<G>d__0.<x>5__1""
+          IL_009a:  ldfld      ""int Test.<G>d__0.<x>5__2""
```

Comment:
Why are the indices larger? If we're reusing locals wouldn't we expect smaller numbers?

Reply:
We delay creating them until we enter scope now. Synthetic things that created later get larger numbers from the disambiguation counter.